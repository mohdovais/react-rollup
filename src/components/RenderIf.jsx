import React from 'react';
import PropTypes from 'prop-types';

RenderIf.propTypes = {
    condition: PropTypes.bool,
    children: PropTypes.array
};

export default class RenderIf extends React.PureComponent {

    render() {
        if (this.props.condition) {
            return this.props.children;
        } else {
            return null;
        }
    }
}
