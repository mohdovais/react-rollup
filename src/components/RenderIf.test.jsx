import React from 'react';
import ReactDOM from 'react-dom';
//import renderer from 'react-test-renderer';
import { act } from 'react-dom/test-utils';
import RenderIf from './RenderIf';

let container;

beforeEach(() => {
    container = document.createElement('div');
    document.body.appendChild(container);
});

afterEach(() => {
    document.body.removeChild(container);
    container = null;
});

test('RenderIf should return null if condition is false', () => {
    act(() => {
        ReactDOM.render(<RenderIf />, container);
    });

    expect(container.innerHTML).toBe('');
});

/*
test('RenderIf should return null if condition is falsy', () => {
    const component = renderer.create(
        <RenderIf condition={null}>Hello</RenderIf>
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
    expect(tree).toBeNull();
});

test('RenderIf should render children if condition is true', () => {
    const component = renderer.create(
        <RenderIf condition={true}>Hello</RenderIf>
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
    expect(tree).toBe('Hello');
});

test('RenderIf should render children if condition is truthy', () => {
    const component = renderer.create(<RenderIf condition={5}>Hello</RenderIf>);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
    expect(tree).toBe('Hello');
});
*/
