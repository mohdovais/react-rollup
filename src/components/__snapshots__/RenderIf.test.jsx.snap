// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RenderIf should render children if condition is true 1`] = `"Hello"`;

exports[`RenderIf should render children if condition is truthy 1`] = `"Hello"`;

exports[`RenderIf should return null if condition is false 1`] = `null`;

exports[`RenderIf should return null if condition is falsy 1`] = `null`;
